Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -43,71 +57,72 @@
     },
 
     {
-       "name": "describeInstanceRootVolume",
-       "action": "aws:executeAwsApi",
-       "onFailure": "Abort",
-       "timeoutSeconds": 60,
-       "maxAttempts": 10,
-       "inputs": {
-         "Service": "ec2",
-         "Api": "DescribeVolumes",
-         "Filters": [
-           {
-              "Name": "attachment.instance-id",
-              "Values": [
-                "{{instanceId}}"
-              ]
-           },
-           {
-              "Name": "attachment.device",
-              "Values": [
-                "{{describeInstance.rootDeviceName}}"
-              ]
-           }
-         ]
-       },
-       "outputs": [
-         {
-            "Name": "rootDeviceVolumeId",
-            "Selector": "$.Volumes[0].Attachments[0].VolumeId",
-            "Type": "String"
-         },
-         {
-            "Name": "rootDeviceVolumeType",
-            "Selector": "$.Volumes[0].VolumeType",
-            "Type": "String"
-         },
-         {
-            "Name": "RootDeviceDeleteOnTermination",
-            "Selector": "$.Volumes[0].Attachments[0].DeleteOnTermination",
-            "Type": "Boolean"
-         }
-       ],
-       "isCritical": true
+      "name": "describeInstanceRootVolume",
+      "action": "aws:executeAwsApi",
+      "onFailure": "Abort",
+      "timeoutSeconds": 60,
+      "maxAttempts": 10,
+      "isCritical": true,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "DescribeVolumes",
+        "Filters": [
+          {
+            "Name": "attachment.instance-id",
+            "Values": [
+              "{{instanceId}}"
+            ]
+          },
+          {
+            "Name": "attachment.device",
+            "Values": [
+              "{{describeInstance.rootDeviceName}}"
+            ]
+          }
+        ]
+      },
+      "outputs": [
+        {
+          "Name": "rootDeviceVolumeId",
+          "Selector": "$.Volumes[0].Attachments[0].VolumeId",
+          "Type": "String"
+        },
+        {
+          "Name": "rootDeviceVolumeType",
+          "Selector": "$.Volumes[0].VolumeType",
+          "Type": "String"
+        },
+        {
+          "Name": "RootDeviceDeleteOnTermination",
+          "Selector": "$.Volumes[0].Attachments[0].DeleteOnTermination",
+          "Type": "Boolean"
+        }
+      ]
     },
 
     {
+      "name": "createSnapshot",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 1800,
+      "onFailure": "Abort",
       "maxAttempts": 3,
       "inputs": {
         "DocumentName": "AWS-CreateSnapshot",
         "RuntimeParameters": {
           "VolumeId": "{{describeInstanceRootVolume.rootDeviceVolumeId}}"
         }
-      },
-      "name": "createSnapshot",
-      "action": "aws:executeAutomation",
-      "timeoutSeconds": 3600,
-      "onFailure": "Abort"
+      }
     },
+
     {
+      "name": "extractSnapshotId",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort",
```

Comment:
Are you aborting because you won't be able to get the snapshot ID, therefore you won't be able to delete the snapshot? Just asking to confirm the logic behind your decision.

Reply:
yes that's correct.